torch/csrc/jit/codegen/cuda/index_compute.h

```diff
@@ -242,6 +232,32 @@ class Index {
       const std::vector<kir::ForLoop*>& loops,
       const std::vector<bool>& root_contiguity,
       bool unswitch = false);
+
+  //! Take a consumer tensorview and loop nest and generates predicats
+  //! associated with the concrete roots of the loop nest. Returns a list of
+  //! predicates, and a list of concrete roots they're associated with. It is
+  //! assumed that no predicate is required if index[i] is an index directly
+  //! from a for loop. This will not catch all cases if we actually have static
+  //! size information for example:
+  //!
+  //! TV[I].split(4)
+  //! would produce the code:
+  //! for(i : I/4)
+  //!   for(j : 4)
+  //!     if( i * 4 + j < TV.size(0))
+  //!       TV[i * 4 + j]...
+  //!
+  //! However if we had TV.size[0] = 16 at "compile time" then we wouldn't need
+  //! the predicate. However we will still generate: for(i : 4) for(j : 4) if( i
+  //! * 4 + j < TV.size(0)) TV[i * 4 + j]...
```

Comment:
`canOmitPredicate` should detect such predicates.

Reply:
Ah okay, copied this over from predicate compute, will remove.